src/redux/products/products.reducer.js

```diff
@@ -0,0 +1,171 @@
+import {
+  SET_PRODUCT,
+  SET_ALL_PRODUCTS,
+  SET_ALL_FILTER_DATA,
+  SET_SORT_BY_PRICE,
+  SET_SORT_BY_DATE,
+  SET_SORT_BY_RATE,
+  SET_SORT_BY_POPULARITY,
+  SET_PRODUCTS_LOADING,
+  SET_CATEGORY_FILTER,
+  SET_PRICE_FILTER,
+  SET_COLORS_FILTER,
+  SET_PATTERNS_FILTER,
+  SET_MODELS_FILTER,
+  SET_SEARCH,
+  SET_PRODUCT_LOADING,
+  SET_PRODUCT_TO_SEND,
+  CLEAR_PRODUCT_TO_SEND,
+  SET_PRODUCTS_ERROR
+} from './products.types';
+
+export const initialState = {
+  productLoading: false,
+  loading: true,
+  sorting: {
+    sortByPrice: 0,
+    sortByRate: 0,
+    sortByPopularity: -1
+  },
+  filters: {
+    categoryFilter: [],
+    modelsFilter: [],
+    colorsFilter: [],
+    patternsFilter: [],
+    searchFilter: ''
+  },
+  filterData: [],
+  selectedroduct: null,
+  products: [],
+  pagesCount: 1,
```

Comment:
Why do we need `pagesCount` here if I cannot find any action that sets it?

Reply:
There is an action setPagesCount in saga `handleFilterLoad `